### ISSUE TITLE
Empty object not reflected in generated wrapper

### ISSUE THREAD

**Issue by carueda** [carueda/tscfg#309]
**Describe the bug**

Empty object not properly handled.

**To Reproduce**

Basic repro steps:

1. Input config schema (say in a file `empty0.spec.conf`):

     ```hocon
     emptyObj : {}
     ```
2. Generate wrapper as usual, eg:
   `java -jar tscfg-x.y.z.jar --spec empty0.spec.conf --scala --cn Empty0`

**Expected behavior**

An `EmptyObj` class included in the generated code, to reflect the input empty object `emptyObj`:
```scala
final case class Empty0(
  emptyObj : Empty0.EmptyObj
)

object Empty0 {
  final case class EmptyObj(
  )
  ...
)
```
However, the following is generated:
```scala
final case class Empty0(
)
...
```

**Additional context/comments**

An empty object per se might not be a typical use case, but the issue becomes significant when using it in combination with the `extends` mechanism. Here's a more realist example:

```hocon
#@define abstract
SomeAbstract {
  something: string
}

#@define extends SomeAbstract
SomeExtension {
}

foo: SomeExtension
```

As in the simpler case above, the `SomeExtension` definition also gets lost, but in this case actually breaking the intended behavior, which is to have a class `SomeExtension` with the members obtained from the extended class!

@sebastian-peter – just fyi as I'm also just noting this issue seems to be affecting some of your projects. Sorry for the trouble. Time has been very scarce for me for a good while, but I still have plans for tscfg in the future, but for now hoping to address this issue soonish.

**Comment by carueda** [comment]
@ckittl @johanneshiry @sebastian-peter @t-ober – sorry for the mass tagging, but I continue to appreciate your use of tscfg for Simona (great use case!), and wanted to bring to your attention this fix, now included in the just released tscfg 1.1.5, which should also fix the closely related issue of resolving string type instead of `@define`.

Along with own unit tests in this repo, I have also tested it with Simona's [config-template.conf](https://github.com/ie3-institute/simona/blob/30c4c95a9fa0842b0250e8251b518fee786a469f/src/main/resources/config/config-template.conf), both in that current form and also with the removal of the "tricks" you have put in place due to the bad handling of empty objects (see diff below).

BTW, I understand you’ve been considering an alternative configuration strategy for Simona, which is entirely reasonable. Unfortunately, tscfg development has become somewhat ad hoc and sporadic in recent years, especially as new features were added—sometimes without much elegance. That said, if you're still interested, I’ve begun working on a new design and cleaner implementation in a private repository. The goal is to make maintenance, improvements, and contributions significantly easier. However, I’m not sure when I’ll be able to integrate these changes into the main codebase. You’ve been fantastic contributors to tscfg, and I’d be delighted to share the repo (and ideas) with anyone interested.

---

```diff
index 6c9d88671..48f584ceb 100644
--- a/src/main/resources/config/config-template.conf
+++ b/src/main/resources/config/config-template.conf
@@ -30,34 +30,24 @@ LoadRuntimeConfig {

 #@define extends BaseRuntimeConfig
 FixedFeedInRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise FixedFeedInRuntimeConfig is handled as String
 }

 #@define extends BaseRuntimeConfig
 PvRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise PvRuntimeConfig is handled as String
 }

 #@define extends BaseRuntimeConfig
 WecRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise WecRuntimeConfig is handled as String
 }

 #@define extends BaseRuntimeConfig
 EvcsRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise EvcsRuntimeConfig is handled as String
   chargingStrategy: String | "maxPower" # can be one of maxPower, constantPower, gridOrientedScheduling or marketOrientedScheduling
   lowestEvSoc: Double | 0.2 # Defines the lowest possible state of charge (SoC) that an EV is allowed to uncharge in vehicle to grid (V2G) mode
 }

 #@define extends BaseRuntimeConfig
 StorageRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise StorageRuntimeConfig is handled as String
   initialSoc: Double | 0 # Defines initial soc of storages as a share of capacity, 0-1
   #@optional
   targetSoc: Double
@@ -65,20 +55,12 @@ StorageRuntimeConfig {

 #@define extends BaseRuntimeConfig
 EmRuntimeConfig {
-  # # # # #
-  # ATTENTION: calculateMissingReactivePowerWithModel and scaling is ignored here.
-  # Cleaner solution is possible with different config framework.
-  # # # # #
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise EmRuntimeConfig is handled as String
   curtailRegenerative: Boolean | false
   aggregateFlex: String | "SELF_OPT_EXCL_REG"
 }

 #@define extends BaseRuntimeConfig
 HpRuntimeConfig {
-  baseRuntimeConfig: BaseRuntimeConfig # this entry is ignored by the config generator,
-  # but cannot removed bc otherwise HpRuntimeConfig is handled as String
 }
```

**Comment by rkrzewski** [comment]
I would love a https://github.com/timvlaer/tscfg-maven-plugin release using tscfg with this fix. Unfortunately development of the maven plugin stalled years ago. @norrisjeremy was working on a rewrite but the PR was rejected https://github.com/timvlaer/tscfg-maven-plugin/pull/5#issuecomment-1538232563. The project was not published AFAICT and the forked repository is gone. @timvlaer, is there any chance for a new release?
